src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -120,5 +120,93 @@ public static async Task<TResult> TryAsyncResolve<TResult>(this IResolveFieldCon
                 }
             }
         }
+
+        private static readonly object _lockExtensions = new object();
+        private static readonly char[] _separators = new char[] { '.' };
+
+        /// <summary>
+        /// Thread safe method to get value by path (key1.key2.keyN) from extensions dictionary.
+        /// </summary>
+        /// <param name="context">Context with extensions response map.</param>
+        /// <param name="path">Path to value in key1.key2.keyN format.</param>
+        /// <returns>Value, if any exists on the specified path, otherwise <c>null</c>.</returns>
+        public static object GetExtension(this IResolveFieldContext context, string path)
+        {
+            if (context == null)
+                throw new ArgumentNullException(nameof(context));
+
+            if (context.Extensions == null || context.Extensions.Count == 0)
+                return null;
+
+            lock (_lockExtensions)
```

Comment:
```suggestion
            lock (context.Extensions)
```
Apparently `IDictionary<TKey, TValue>` doesn't inherit from `ICollection` and does not provide `SyncRoot`